CommonTools/ParticleFlow/python/pfTaus_cff.py

```diff
@@ -28,75 +28,76 @@
 
 # PiZeroProducers
 
-pfJetsLegacyHPSPiZeros = ak4PFJetsLegacyHPSPiZeros.clone()
-pfJetsLegacyHPSPiZeros.jetSrc = cms.InputTag("ak4PFJets")
-
+pfJetsLegacyHPSPiZeros = ak4PFJetsLegacyHPSPiZeros.clone(
+    jetSrc = "ak4PFJets"
+)
 pfTauPFJetsRecoTauChargedHadrons = ak4PFJetsRecoTauChargedHadrons.clone()
 
-pfTauTagInfoProducer = pfRecoTauTagInfoProducer.clone()
-pfTauTagInfoProducer.PFCandidateProducer = jetConfig.ak4PFJets.src
-pfTauTagInfoProducer.PFJetTracksAssociatorProducer = 'pfJetTracksAssociatorAtVertex'
-
+pfTauTagInfoProducer = pfRecoTauTagInfoProducer.clone(
+    PFCandidateProducer = jetConfig.ak4PFJets.src ,
+    PFJetTracksAssociatorProducer = 'pfJetTracksAssociatorAtVertex'
+)
 # Clone tau producer
-pfTausProducer = hpsPFTauProducer.clone()
+pfTausProducer = hpsPFTauProducer.clone(
+    src = "pfTausProducerSansRefs"
+)
 pfTausCombiner = combinatoricRecoTaus.clone(
-    jetSrc = "ak4PFJets",
-    piZeroSrc = "pfJetsLegacyHPSPiZeros",
+    jetSrc           = "ak4PFJets",
+    piZeroSrc        = "pfJetsLegacyHPSPiZeros",
     chargedHadronSrc = "pfTauPFJetsRecoTauChargedHadrons"
 )
 for mod in pfTausCombiner.modifiers:
     if mod.name == "TTIworkaround": mod.pfTauTagInfoSrc = "pfTauTagInfoProducer"
 
-pfTausSelectionDiscriminator = hpsSelectionDiscriminator.clone()
-pfTausSelectionDiscriminator.PFTauProducer = cms.InputTag("pfTausCombiner")
-pfTausProducerSansRefs = hpsPFTauProducerSansRefs.clone()
+pfTausSelectionDiscriminator = hpsSelectionDiscriminator.clone(
+    PFTauProducer = "pfTausCombiner"
+)
+pfTausProducerSansRefs = hpsPFTauProducerSansRefs.clone(
```

Comment:
This is a duplicate of the explicit configuration three lines below (a duplicate that was already present in the original config).
The two configurations look different, the explicit one being probably the correct one:  which is the one picked up by python?
@mbluj @swozniewski please check and suggest which one of the two instances of `pfTausProducerSansRefs` to remove (or suggest otherwise, in case I misunderstood the usage of this duplication)

Reply:
In our understanding python takes most recent definition of an object, so here it is the explicit one.
The explicit definition indeed differs a bit from the first one which is a clone of Tau POG default (modulo different input). We guess that it was an intended choice to define it differently, so we suggest to keep only the explicit one. However, maybe owners of this configuration should bless the choice if it is possible to identify them (*)?

Technical suggestion: it could be nicer to have a clone definition with settings as the present in the explicit one? It can either use default Tau POG configuration to clone or one from default cfi (`RecoTauTag.RecoTau.recoTauCleaner_cfi`) It will look more or less like this:
```
pfTausProducerSansRefs = hpsPFTauProducerSansRefs.clone(
    src = "pfTausCombiner",
    outputSelection = "",
    verbosity = 0,
    cleaners = [
        cleaners.unitCharge,
        cms.PSet(
            name = cms.string("leadStripPtLt2_5"),
            plugin = cms.string("RecoTauStringCleanerPlugin"),
            tolerance = cleaners.tolerance_default,
            selection = cms.string("signalPiZeroCandidates().size() = 0 | signalPiZeroCandidates()[0].pt > 2.5"),
            selectionPassFunction = cms.string("0"),
            selectionFailValue = cms.double(1e3)
        ),
        cms.PSet(
            name = cms.string("HPS_Select"),
            plugin = cms.string("RecoTauDiscriminantCleanerPlugin"),
            tolerance = cleaners.tolerance_default,
            src = cms.InputTag("pfTausSelectionDiscriminator"),
        ),
        cleaners.combinedIsolation
    ]
)
```

(*) Is this config a part of PF2PAT? So, it is owned by PFlow or XPOG?